sabnzbd/nzbstuff.py

```diff
@@ -371,15 +373,72 @@ def __init__(self, date, subject, raw_article_db, file_bytes, nzo):
             # Any articles left?
             if raw_article_db:
                 # Save the rest
-                sabnzbd.save_data(raw_article_db, self.nzf_id, nzo.admin_path)
+                self.finish_import(raw_article_db)
+                self.pickle_articles(initial_pickle=True)
             else:
                 # All imported
                 self.import_finished = True
 
-    def finish_import(self):
-        """ Load the article objects from disk """
+    def pickle_articles(self, initial_pickle=False):
+        """ Pickle articles to file """
+        if not initial_pickle:
+            if (
+                not self.import_finished
+                # Don't pickle tiny files
+                or len(self.articles) < 3
+                # Don't pickle active files
+                or self.pickle_lock_time > time.time()
+                or len(self.articles) != len(self.decodetable)
+            ):
+                return False
+            # Don't pickle active files
+            for article in self.articles:
+                if article.fetcher:
+                    return False
+
+        logging.debug("pickle %s", self.filename)
+        first_article = self.decodetable.pop(0)
```

Comment:
Why do we need to separate the first article again? I know there was a reason for it.

Reply:
The completion status check only checks the first article, and it has to always be imported because the import status is not checked before trying to read it.